pkg/ccl/backupccl/backup_test.go

```diff
@@ -3377,6 +3377,53 @@ func TestBackupTenantsWithRevisionHistory(t *testing.T) {
 	require.Contains(t, fmt.Sprint(err), msg)
 }
 
+// TestBackupJobFailsInRestoredTenant asserts that a backup job in a restored
+// tenant fails on resume, even if the backup job details are not fully
+// populated. This test specifically addresses the bug described in
+// https://github.com/cockroachdb/cockroach/issues/112114
+func TestBackupJobFailsInRestoredTenant(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	const numAccounts = 1
+	ctx := context.Background()
+	tc, systemDB, _, cleanupFn := backupRestoreTestSetup(t, singleNode, numAccounts, InitManualReplication)
+	defer cleanupFn()
+
+	t10, err := tc.Servers[0].TenantController().StartTenant(ctx, base.TestTenantArgs{
+		TenantID: roachpb.MustMakeTenantID(10),
+		TestingKnobs: base.TestingKnobs{
+			JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals()},
+	})
+	require.NoError(t, err)
+	t10Sql := sqlutils.MakeSQLRunner(t10.SQLConn(t, ""))
+	potentialPausePoints := []string{
+		"backup.after.write_first_checkpoint",
+		"backup.after.details_has_checkpoint",
+	}
+	rng, _ := randutil.NewTestRand()
+	pausePoint := potentialPausePoints[rng.Intn(2)]
+	t10Sql.Exec(t, fmt.Sprintf("SET CLUSTER SETTING jobs.debug.pausepoints = '%s'", pausePoint))
+	_, err = t10.SQLConn(t, "").Exec("BACKUP INTO 'nodelocal://1/example-schedule'")
```

Comment:
I think you could use `t10sql.ExpectErr`

Reply:
done.